swingset/src/main/java/com/nqadmin/swingset/SSDataNavigator.java

```diff
@@ -704,7 +694,10 @@ private void addNavListeners() {
 			logger.debug("ADD button clicked.");
 			removeRowsetListener();
 			try {
-				if (!commitChanges()) return;
+				// Only attempt to commit to database if we are on a row (non-empty rowset)
+				if (rowSet.getRow() > 0) {
+					if (!commitChangesToDatabase(true)) return;
+				}
```

Comment:
These three lines can be replaced by
```
commitChangesToDatabase(true);
```
No conditional, it's basically a nop when `rowSet.getRow() == 0`.

I had trouble figuring out what was going on, why wasn't it ok. I guess this is an example of where treating empty data set the same as `dBNav.allowUpdate()` is an issue. After further thought, I'm wondering if
```
if (!dBNav.allowUpdate()) return;
```
should be in this method. Previously, if `dBNav.allowUpdate()` was false, the insert row could not be opened.

Reply:
Agree. That row check became redundant with recent changes. Pushed the change.

dBNav.allowUpdate() is true by default. Guess there could be instances where a developer overrides it to false (which is presumably why the SSDBNav class is there), but that class is meant to be actually stop the user from doing things whereas the get/setModification() method and class variable allow the user to proceed as though things are actually persisted to the database (but they aren't). I think of modification==false as being sort of a demo mode.

Reply:
So I can't find anything else to nit pick about :-)

I finally looked at SSDBNav, I misunderstood some things. I still don't entirely understand it, but I have a better idea.

There's the perform ops, they're pretty clear; IIRC they're for workarounds for problems with databases. It seems they wouldn't change after app startup and connecting with the database.

Then there's the allow* methods. I'm confused how they might be used. I was wondering if they might be for dynamic changes depending on things that are going on. Or are they an escape hatch for yet unforseen problems?